### PR TITLE
MCSC 1023 Placeholder screens for part 3

### PR DESCRIPTION
**Pull request by mrah93**

### CODE REVIEW

File: app/pages/setupquestions/annualallowance/NetIncomeAbove190KIn2023Page.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+package pages.setupquestions.annualallowance
+
+import models.{LTAKickOutStatus, NormalMode, UserAnswers}
+import pages.QuestionPage
+import play.api.libs.json.JsPath
+import play.api.mvc.Call
+
+case object NetIncomeAbove190KIn2023Page extends QuestionPage[Boolean] {
+
+  override def path: JsPath = JsPath \ toString
```
Comment (ElliotAndersonHMRC) [inline app/pages/setupquestions/annualallowance/NetIncomeAbove190KIn2023Page.scala:26]: override def path: JsPath = JsPath \ "setup" \ "aa" \ toString

File: app/pages/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023Page.scala [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import pages.QuestionPage
+import pages.annualallowance.taxyear.AmountOfGiftAidPage
+import play.api.libs.json.JsPath
+import play.api.mvc.Call
+
+import scala.util.Try
+
+case object PIAAboveAnnualAllowanceIn2023Page extends QuestionPage[Boolean] {
+
+  override def path: JsPath = JsPath \ toString
```
Comment (ElliotAndersonHMRC) [inline app/pages/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023Page.scala:30]: override def path: JsPath = JsPath \ "setup" \ "aa" \ toString

**Review by ElliotAndersonHMRC — CHANGES_REQUESTED** [review]
change paths

File: app/views/setupquestions/annualallowance/NetIncomeAbove190KIn2023View.scala.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        @govukRadios(
+            RadiosViewModel.yesNo(
+                field = form("value"),
+                legend = LegendViewModel(messages("netIncomeAbove190KIn2023.heading")).asPageHeading()
+            )
+        )
+
+        @govukButton(
+            ButtonViewModel(messages("site.continue"))
+        )
```
Comment (MichaelAAbbott) [inline app/views/setupquestions/annualallowance/NetIncomeAbove190KIn2023View.scala.html:44]: Needs our save and return button component

File: app/views/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023View.scala.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        @govukRadios(
+            RadiosViewModel.yesNo(
+                field = form("value"),
+                legend = LegendViewModel(messages("pIAAboveAnnualAllowanceIn2023.heading")).asPageHeading()
+            )
+        )
+
+        @govukButton(
+            ButtonViewModel(messages("site.continue"))
+        )
```
Comment (MichaelAAbbott) [inline app/views/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023View.scala.html:44]: Needs our save and return button component

File: app/controllers/setupquestions/annualallowance/NetIncomeAbove190KIn2023Controller.scala [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          value =>
+            for {
+              updatedAnswers <- Future.fromTry(request.userAnswers.set(NetIncomeAbove190KIn2023Page, value))
+              redirectUrl     = NetIncomeAbove190KIn2023Page.navigate(mode, updatedAnswers).url
+              answersWithNav  = SetupSection.saveNavigation(updatedAnswers, redirectUrl)
+              _              <- userDataService.set(answersWithNav)
+            } yield Redirect(redirectUrl)
+        )
+  }
```
Comment (MichaelAAbbott) [inline app/controllers/setupquestions/annualallowance/NetIncomeAbove190KIn2023Controller.scala:71]: Needs to be able to update kickout statuses to 2 when yes

File: app/controllers/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023Controller.scala [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          value =>
+            for {
+              updatedAnswers <- Future.fromTry(request.userAnswers.set(PIAAboveAnnualAllowanceIn2023Page, value))
+              redirectUrl     = PIAAboveAnnualAllowanceIn2023Page.navigate(mode, updatedAnswers).url
+              answersWithNav  = SetupSection.saveNavigation(updatedAnswers, redirectUrl)
+              _              <- userDataService.set(answersWithNav)
+            } yield Redirect(redirectUrl)
+        )
+  }
```
Comment (MichaelAAbbott) [inline app/controllers/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023Controller.scala:71]: Needs to be able to update kickout statuses to 2 when yes

**Review by MichaelAAbbott — CHANGES_REQUESTED** [review]

File: app/pages/setupquestions/annualallowance/NetIncomeAbove190KIn2023Page.scala (hunk 1 above)
Comment (mrah93) [reply app/pages/setupquestions/annualallowance/NetIncomeAbove190KIn2023Page.scala]: Done

File: app/pages/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023Page.scala (hunk 2 above)
Comment (mrah93) [reply app/pages/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023Page.scala]: Done

File: app/views/setupquestions/annualallowance/NetIncomeAbove190KIn2023View.scala.html (hunk 3 above)
Comment (mrah93) [reply app/views/setupquestions/annualallowance/NetIncomeAbove190KIn2023View.scala.html]: Done

File: app/views/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023View.scala.html (hunk 4 above)
Comment (mrah93) [reply app/views/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023View.scala.html]: Done

File: app/controllers/setupquestions/annualallowance/NetIncomeAbove190KIn2023Controller.scala (hunk 5 above)
Comment (mrah93) [reply app/controllers/setupquestions/annualallowance/NetIncomeAbove190KIn2023Controller.scala]: Completely forgot about statuses. Added it now

**Review by ElliotAndersonHMRC — COMMENTED** [review]
paths resolved

**Review by ElliotAndersonHMRC — APPROVED** [review]
paths resolved

File: app/views/setupquestions/annualallowance/NetIncomeAbove190KIn2023View.scala.html (hunk 3 above)
Comment (MichaelAAbbott) [reply app/views/setupquestions/annualallowance/NetIncomeAbove190KIn2023View.scala.html]: See here for our custom govukbutton

https://github.com/hmrc/calculate-public-pension-adjustment-frontend/blob/8d089b2ba0a1c7aa5eded6ca7f3b9173282cc7d8/app/views/annualallowance/taxyear/AddAnotherSchemeView.scala.html#L50C7-L50C26

File: app/views/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023View.scala.html (hunk 4 above)
Comment (MichaelAAbbott) [reply app/views/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023View.scala.html]: See here for our custom govukbutton

https://github.com/hmrc/calculate-public-pension-adjustment-frontend/blob/8d089b2ba0a1c7aa5eded6ca7f3b9173282cc7d8/app/views/annualallowance/taxyear/AddAnotherSchemeView.scala.html#L50C7-L50C26

File: app/controllers/setupquestions/annualallowance/NetIncomeAbove190KIn2023Controller.scala [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          value => {
+            val revisedPSSStatus = request.userAnswers.get(SavingsStatementPage).get
+            val aaKickOutStatus  = if (value && revisedPSSStatus) kickOutStatusCompleted else kickOutStatusFalse
```
Comment (MichaelAAbbott) [inline app/controllers/setupquestions/annualallowance/NetIncomeAbove190KIn2023Controller.scala:67]: These pages don't rely on PSS Status to determine kickout status I believe, double check mural, should just be determined based on the page value entered by the user.

File: app/controllers/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023Controller.scala [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          value => {
+            val revisedPSSStatus = request.userAnswers.get(SavingsStatementPage).get
+            val aaKickOutStatus  = if (value && revisedPSSStatus) kickOutStatusCompleted else kickOutStatusFalse
```
Comment (MichaelAAbbott) [inline app/controllers/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023Controller.scala:67]: These pages don't rely on PSS Status to determine kickout status I believe, double check mural, should just be determined based on the page value entered by the user.

**Review by MichaelAAbbott — CHANGES_REQUESTED** [review]

File: app/controllers/setupquestions/annualallowance/NetIncomeAbove190KIn2023Controller.scala (hunk 7 above)
Comment (mrah93) [reply app/controllers/setupquestions/annualallowance/NetIncomeAbove190KIn2023Controller.scala]: My mistake copy and paste error

File: app/controllers/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023Controller.scala (hunk 8 above)
Comment (mrah93) [reply app/controllers/setupquestions/annualallowance/PIAAboveAnnualAllowanceIn2023Controller.scala:67]: My mistake copy and paste error

**Review by MichaelAAbbott — APPROVED** [review]

**Pull request merged**